docs/src/proposals/optimistic-confirmation-and-slashing.md

```diff
@@ -0,0 +1,89 @@
+# Optimistic Confirmation and Slashing
+
+Progress on optimistic confirmation can be tracked here
+
+https://github.com/solana-labs/solana/projects/52
+
+At the end of May, the mainnet-beta is moving to 1.1, and testnet is moving
+to 1.2. With 1.2, testnet will behave as if we have 1-block conf as
+long as no more than 4.66% of the validators are acting maliciously.
+Applications can assume that 2/3+ votes observed in gossip confirm a
+block or that at least 4.66% of the network is violating the protocol.
+
+## How does it work?
+
+The general idea is that validators have to continue voting, following
+their last fork, unless they can construct a proof that their fork
+may not reach finality. The way validators construct this proof is by
+collecting votes for all the other forks, excluding their own. If
+the set of valid votes represents over 1/3+X of the epoch stake weight,
+there is no way for the validators current fork to reach 2/3+ finality.
```

Comment:
I wouldn't say there's `no way`, it's possible, it's possible that some of the `1/3+X` could switch over. It's just that there's also a nonzero possibility that another fork outside of the current one can be finalized, hence why a switch is allowed.